### ISSUE TITLE
Certain items do not compress/decompress in limited barrels

### ISSUE THREAD

**Issue by Kaligraphy22** [P3pp3rF1y/SophisticatedStorage#330]
Launcher: Prism Launcher
Minecraft version: 1.20.1
Forge version: Neoforge 47.1.79
Sophisticated Core version: sophisticatedcore-1.20.1-0.5.108.504
Sophisticated Storage version: sophisticatedstorage-1.20.1-0.8.58.644

--Issue--
Added regular minecraft clay to a limited barrel II that has a compression upgrade on it. The barrell will only accept clay or clay blocks, but not both at the same time. It does the same thing with minecraft snowballs, prismarine shards and melons.

It also does not accept the Supplementaries recipe of a block of flint (9 flint, 3x3 recipe) with a compression upgrade.

As I was writing this post, I tested another method. I put the items in the barrell II before adding the compression upgrade. The items were accepted but did not compress or decompress as items were added or removed. The exception, however, was the block of flint recipe added by Supplementaries. It's the only on that worked with this method.

**Comment by P3pp3rF1y** [comment]
I am pretty sure that most if not all can't be converted both ways using recipes? do you actually see recipe for both 4xclay to clay block and clayblock to 4xclay? And the same for the rest of these?

Compression will only work for those that have recipes for both ways or you can use a config value to add custom compressions like there's a case for quartz by default.
Which I guess I could potentially add clay as well not so sure about snow balls but probably those as well.

**Comment by Kaligraphy22** [comment]
You know, I never thought to check to see if there was a receipe to to uncompress those blocks. There isn't. That would explain my issue. I assumed the compression upgrade acted much like the compacting drawers from Storage Drawers. Those compacting drawers work on clay, snow and both sandstones and sands despite not having a recipe to undo the compression.

**Comment by P3pp3rF1y** [comment]
there's a config `additionalDecompressibleItems = ["minecraft:glowstone=4xminecraft:glowstone_dust", "minecraft:quartz_block=4xminecraft:quartz"]` and as you can see there are a few things added in there by default which make quartz and glowstone decompressible. I guess I will add the other stuff as default in config so that there's no confusion when people are switching between drawers and soph storage

**Comment by Kaligraphy22** [comment]
Oh, so I can add the sands and stuff myself to that config? And I'd be good to go?

**Comment by P3pp3rF1y** [comment]
Yes that's what config are for. I will add more defaults there as well

**Comment by BooBooFastCat1** [comment]
when i try to add additional compression recipes via the config it does not do anything

**Comment by P3pp3rF1y** [comment]
> when i try to add additional compression recipes via the config it does not do anything

Config only adds decompression recipes it doesn't add both ways - it's only meant for something that can be compressed, but not decompressed by default like let's say quartz to quartz block. You should be adding recipes both ways if the item in question doesn't even have a compression. And if you were only trying to add decompression I would need specifics rather than saying it just doesn't work.

**Comment by BooBooFastCat1** [comment]
i try adding decompression for ice packed ice and blue ice with "minecraft:packed_ice=9xminecraft:ice", "minecraft:blue_ice=9xminecraft:packed_ice" in the config but when i try to put any of them in the drawer it does not allow compressing

**Comment by P3pp3rF1y** [comment]
@BooBooFastCat1 I just copy pasted your settings into my config and it just works. though I noticed that if these were already cached the cache doesn't get cleared when a world is loaded. So perhaps you tried to change these with game already running in which case you should restart the game (also make sure that the settings are still there before loading into the world sometimes configs can be a bit funky when changed while the game is running)

**Comment by BooBooFastCat1** [comment]
thanks having the game open was problem

**Comment by P3pp3rF1y** [comment]
I have added the additional decompressible item config defaults for parity with storage drawers in the latest release
